python/py-matplotlib/Portfile

```diff
@@ -5,7 +5,7 @@ PortGroup           python 1.0
 
 name                py-matplotlib
 version             3.1.1
-revision            1
+revision            2
```

Comment:
It is not necessary to increase the `revision` because of removing a dependency. Nothing changes for anyone who has it currently built but it will trigger a rebuild.

Reply:
Yes, and I think that's what we want here... right? At least in my understanding and local testing that's the only way to have MacPorts understand that ```py-pyobjc``` isn't a dependency anymore. If that's indeed true the ```revision``` should also be increased for the other two occurences in the Portfile.

Reply:
Adding a dependency only tells port to install py-pyobjc. So everyone who has py-matplotlib installed also has py-pyobjc installed. Increasing the revision does not change that. So it is not necessary.

Reply:
Right, but we are removing the py-pyobjc-cocoa dependency. So whoever has py-matplotlib installed, will have py-pyobjc-cocoa listed now as a dependency (in the registry?). Only removing the dependency from the py-matplotlib Portfile does not help as the user still cannot uninstall without a warning about breaking py-matplotlib.

```
>>> sudo port uninstall py37-pyobjc-cocoa
Note: It is not recommended to uninstall/deactivate a port that has dependents as it breaks the dependents.
The following ports will break: py37-matplotlib @3.1.1_1
```

As far as I can tell, only increasing the revision and rebuilding the port without the py-pyobjc-cocoa dependency listed will solve that problem.

Reply:
The warning is wrong in this case and the user can still uninstall it using `-f`. Most people won't even care about this. But for the ones that do, they can uninstall it.

Reply:
The MacPorts policy has always been to increase the revision of a port when the dependencies change.
This pull request should be merged if there are no other concerns.

Reply:
I have another PR that obsoletes the `py-pyobjc-cocoa` port, folds it into `py-pyobjc`, and marks it as replaced by it. I rolled back an earlier version of that change due to issues with dependent ports, so please keep this in place. With binary archives, a revision bump really doesn’t affect users all that much.

Reply:
I am terribly sorry, but could you clarify what you mean by "please keep this in place?"
Are you asking to hold off on merging?

> a revision bump really doesn’t affect users all that much

Again, I am sorry if I don't understand, but the the MacPorts policy has always been to increase the revision of a port when the dependencies change.
But that is exactly what your PR does, so isn't this PR in good shape?

Reply:
Apologies for the confusion; I meant to keep the change in the PR as-is, and that I believe it’s safe to apply 🙂

Reply:
> Apologies for the confusion; I meant to keep the change in the PR as-is, and that I believe it’s safe to apply 🙂

I entirely agree.

Reply:
@MarcusCalhoun-Lopez @danchr as you can see from the discussion above I agree with increasing the ```revision```. So please update your PR to increase the ```revision``` for the two other occurrences in the Portfile and I will merge it.

Reply:
> @MarcusCalhoun-Lopez @danchr as you can see from the discussion above I agree with increasing the `revision`. So please update your PR to increase the `revision` for the two other occurrences in the Portfile and I will merge it.

Ah, forgot about that, sorry! Done!